pkg/process/monitor_linux.go

```diff
@@ -0,0 +1,259 @@
+//go:build linux
+// +build linux
+
+/*
+   Copyright The nydus Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package process
+
+import (
+	"net"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/nydus-snapshotter/pkg/errdefs"
+	"github.com/containerd/nydus-snapshotter/pkg/utils/retry"
+)
+
+// LivenessMonitor liveness of a nydusd daemon.
+type LivenessMonitor interface {
+	// Subscribe death event of a nydusd daemon.
+	// `path` is where the monitor is listening on.
+	Subscribe(id string, path string, notifier chan<- deathEvent) error
```

Comment:
I think it's clean if using `Subscribe(id string, path string, func() error) error`, and we can remove `handleDaemonDeathEvent()`. It's not a blocker for this PR.

Reply:
Then it has to bring a lot of improvements here. I am suggesting pending the improvement and discuss it later. Do you agree?

Reply:
agree.